### PR TITLE
Use Time_t for ParticipantLocationTopic

### PR DESCRIPTION
**Pull request by jrw972**

### CODE REVIEW

File: dds/DCPS/RTPS/Spdp.cpp [hunk 1]
```diff
@@ -434,40 +434,40 @@ Spdp::process_location_updates_i(DiscoveredParticipantIter iter)
...
-    const unsigned long expr = now - rtps_duration_to_time_duration(
-                                                                    iter->second.pdata_.leaseDuration,
-                                                                    iter->second.pdata_.participantProxy.protocolVersion,
-                                                                    iter->second.pdata_.participantProxy.vendorId).value().sec();
-    if ((location_data.location & DCPS::LOCATION_LOCAL) && location_data.local_timestamp < expr) {
+    const DDS::Time_t expr = (now - rtps_duration_to_time_duration(
+                                                                   iter->second.pdata_.leaseDuration,
+                                                                   iter->second.pdata_.participantProxy.protocolVersion,
+                                                                   iter->second.pdata_.participantProxy.vendorId)).to_dds_time();
+    if ((location_data.location & DCPS::LOCATION_LOCAL) && DCPS::operator<(location_data.local_timestamp, expr)) {
```
Comment (mitza-oci) [inline dds/DCPS/RTPS/Spdp.cpp]: In other parts of the code, the same thing is done with a `using` declaration.

File: dds/DCPS/DiscoveryBase.h [hunk 2]
```diff
@@ -1624,9 +1624,9 @@ namespace OpenDDS {
           std::memcpy(location_data_.guid, &guid, sizeof(guid));
           location_data_.location = 0;
           location_data_.change_mask = 0;
-          location_data_.local_timestamp = 0;
-          location_data_.ice_timestamp = 0;
-          location_data_.relay_timestamp = 0;
+          location_data_.local_timestamp = {0, 0};
```
Comment (mitza-oci) [inline dds/DCPS/DiscoveryBase.h]: C++03 incompatibility?

Comment (jrw972) [reply dds/DCPS/DiscoveryBase.h]: Foiled again.

**Comment by mitza-oci** [comment]
```
ParticipantLocationListener.java:45: error: incompatible types: int cannot be converted to Time_t
          new ParticipantLocationBuiltinTopicData(new byte[16], 0, 0, "", 0, "", 0, "", 0));
```

**Pull request merged**